Change bot replies to reflect new caps

Dracologist opened the pull request:
I decided to try to change the standard investment confirmation to include how many times you've invested in that post and the maximum number of times you can invest in the same post. I also tried to make the rejection message for investments above the maximum amount more informative. 

You might want to be pretty thorough in checking my code if you decide you want to use these because I have limited experience with Python

Dracologist committed:
New bot replies to reflect new investment limits

Dracologist committed:
Making bot replies reflect new caps

Dracologist committed:
Added !invest min and !invest max

I was feeling bold and thought that the new update might make it harder for people to figure out what they can and can't invest, so I decided to try my hand at making it possible to automatically invest the minimum or maximum amount allowed

Dracologist commented:
I decided to try my hand at creating commands to automatically invest the minimum or maximum amount of MemeCoins allowed because why not? If nothing else, it's decent practice for me to learn more about Python

Dracologist committed:
No point in checking for a suffix after max and min

Dracologist committed:
Making it recognize !i as short for !invest

Some people on my firm's discord thought it was a good idea, so I figured it wouldn't hurt anything to write it. After all, you guys can just reject the commit if you don't like it, right?

Dracologist commented:
I'm just adding small quality of life features me and people from my firm think might be cool because this is fun. Feel free to reject any changes you don't like

Dracologist committed:
Fixed !invest max so it doesn't try to invest more than the investor's balance

thecsw commented on src/comment_worker.py:
```diff
@@ -170,7 +170,7 @@ def __call__(self, comment):
         # Parse the comment body for a command
         for reg in self.regexes:
-            matches = reg.fullmatch(comment.body.strip())
+            matches = reg.fullmatch(re.sub(r"![Ii]\s+", "!invest ", comment.body.strip()))
```
Why are you manually setting this here?

thecsw commented on src/message.py:
```diff
@@ -271,6 +274,17 @@ def modify_min_invest(minim):
+        replace("%MAX%", format(int(maxim), ",d"))
+
+
+
```
Too many empty lines!

thecsw reviewed (changes requested):
I like the PR update and making the new feature more agile. So, here is the list of changes I think would be nice:
 - I am not sure we need to have max/min in the regex
 - There are some empty lines in `message.py`

Dracologist committed:
Yeah, my attempt to make !i a thing was clumsy at best

I still think !invest max will be handy to have so you don't have to guess what the highest amount you're allowed to invest in a post is, though. It also just didn't make sense to do !invest max without an equivalent !invest min command, even if it might not see much use

Dracologist committed:
Removing extra blank line

thecsw reviewed (commented):
If we set the limit to 50%, they could do just 50%

Money investing is not a casino where you can just go *max* or *ALL IN*. That is why we never made `!invest all` or `!all`, it just doesn't make sense in a economical situtation. 

If you have thoughts about this, I would love to chat

Dracologist commented:
I'm pretty sure real-life investing doesn't have any limit to how much of your money you're allowed to invest in a single stock either, but adding that limit is probably going to improve r/MemeEconomy for most users. That's kinda my logic behind the `max` thing. It's a lot smaller, but it's convenient for when you feel like putting as much as you can into a meme.

I'll remove it from this branch and make a new branch off of this branch with the max and min

Dracologist committed:
Removed max and min from this branch

hazzakak commented:
What else is wrong with this pr?

thecsw reviewed (approved):
Seems fine to me! This PR is approved. It will be merged if the mods or r/MemeEconomy decide to or when we start a new season with bonding stocks update. Thank you!

thecsw commented:
@HarryjosephDevelopment Everything seems fine. Just approved it.

hazzakak commented:
> @HarryjosephDevelopment Everything seems fine. Just approved it.

Sounds good! Is there a list of updates which will come out in the next season?

thecsw commented:
@HarryjosephDevelopment the dream is to implement the stocks update from #373 and implement the codebase in Go instead of Python #394 

Also, @Keanu73 and I are talking about a possibility of rebuilding the bot with JS, see #394 for more details

luisbg commented:
I like this a lot more than the original PR 😄

luisbg commented:
> Also, @Keanu73 and I are talking about a possibility of rebuilding the bot with JS, see #394 for more details

What are the benefits of rebuilding in JS? I'm curious.

thecsw commented:
Just to clarify, the changes are approved but they *WILL NOT* be merged until the necessity comes.
